src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -130,27 +130,32 @@ def spin():
 
         # Source code compression
         done_spinner = False
-        thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
-        tar_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
-        archive_source_code(tar_file_path, source)
+        source_file_path = source
+        if os.path.isdir(source):
+            thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
+            source_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
+            archive_source_code(source_file_path, source)
         done_spinner = True
         thread.join()
 
         # File upload
         done_spinner = False
-        thread = display_spinner("Uploading compressed data")
+        thread = display_spinner("Uploading data")
         headers = {'Authorization': 'Bearer ' + token}
         try:
-            tar_file = open(tar_file_path, "rb")
-            files = [("file", ("build_data.tar.gz", tar_file, "application/x-tar"))]
+            source_file = open(source_file_path, "rb")
+            file_extension = os.path.splitext(source_file_path)[1]
+            files = [("file", ("build_data" + file_extension, source_file))]
             response_file_upload = requests.post(
                 upload_endpoint,
                 files=files,
                 headers=headers)
         finally:
-            # Close and delete the file now that it was uploaded.
-            tar_file.close()
-            os.unlink(tar_file_path)
+            # Close the file now that it was uploaded.
+            source_file.close()
+            # if customer uploaded source file is a folder, delete the temp compressed file
+            if os.path.isdir(source):
```

Comment:
Nit: please consider creating a variable for `os.path.isdir(source)` so that we're always consistent between the first check and this one, and so that we avoid potential exceptions/slowness when running this.